### PR TITLE
mOSL 2.4.0 (new formula)

### PR DESCRIPTION
**Pull request by 0xmachos**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

mOSL (macOS Lockdown) is my tool which audits/ fixes macOS security settings.

### CODE REVIEW

File: Formula/mosl.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  # Required to valide signature against Lockdown.minisig
+
+  def install
+    bin.install "Lockdown"
+    prefix.install "Lockdown.minisig"
+    # Signature file generated by minisign
+  end
+
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/Lockdown debug | grep 'mOSL Version' | awk '{print $3}' | tr -d 'v'").chomp
```
Comment (zbeekman) [inline Formula/mosl.rb]: We need a better test here:

1. Tests shouldn't pipe out to awk or other unix tools to do text processing
2. The test should actually test some sort of functionality or correctness of the installation

File: Formula/mosl.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Mosl < Formula
+  desc "Bash script to audit and fix macOS Mojave (10.14.x) security settings"
+  homepage "https://github.com/0xmachos/mOSL"
+  url "https://github.com/0xmachos/mOSL/archive/v2.4.0.tar.gz"
+  sha256 "6cf2f301844085f36aa131867a4cc11d7d11a3b04b25c5d3ea35f71c39296eef"
+
+  depends_on "minisign" => :recommended
```
Comment (zbeekman) [inline Formula/mosl.rb]: Please get rid of the `:recommended` here. We have deprecated options, and formulae should be built to be as featureful as possible to benefit as many users as possible.

```suggestion
  depends_on "minisign"
```

**Review by zbeekman — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution to homebrew. Can you please resolve the inline comments, squash and force push?

Thanks!

**Comment by zbeekman** [comment]
Also, the readme states:

```
This script will only ever support the latest macOS release
```

At homebrew we need to support the latest 3 macOS releases. If this formula only supports the latest release then:

1. It may be ineligible for inclusion in Homebrew
2. It should include logic to indicate that it only supports the latest macOS release

File: Formula/mosl.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class Mosl < Formula
+  desc "Bash script to audit and fix macOS Mojave (10.14.x) security settings"
+  homepage "https://github.com/0xmachos/mOSL"
+  url "https://github.com/0xmachos/mOSL/archive/v2.4.0.tar.gz"
+  sha256 "6cf2f301844085f36aa131867a4cc11d7d11a3b04b25c5d3ea35f71c39296eef"
```
Comment (zbeekman) [inline Formula/mosl.rb]:
```suggestion
  sha256 "6cf2f301844085f36aa131867a4cc11d7d11a3b04b25c5d3ea35f71c39296eef"
  
  bottle :unneeded
```

**Review by zbeekman — CHANGES_REQUESTED** [review]

**Comment by zbeekman** [comment]
Looks like a great addition, but a few things need to be addressed first. Please make sure you run `brew audit --strict --online mosl` before pushing.

**Comment by fxcoudert** [comment]
I don't think a tool that has a policy to only support latest macOS is great for Homebrew core. It's probably a better fit for an external tap https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap (especially given it does not appear so widely used).

**Comment by 0xmachos** [comment]
I’m happy to creat my own Tap if the only supporting the current macOS release policy makes it incompatible with core.

**Comment by 0xmachos** [comment]
It might in the future, I'm looking at rewriting it but I don't have an eta for doing that.

Feel free to close this, I've setup a tap at [0xmachos/homebrew-mOSL](https://github.com/0xmachos/homebrew-mOSL) and I'll have a look at  `interesting taps`.

Thanks for your feedback.

**Pull request closed without merging**